### ISSUE TITLE
[Bug] strikethrough for checked indented task list elements is overlapping on the left

### ISSUE THREAD

**Issue by leo-hstone** [Zettlr/Zettlr#3029]
### Description

As you can see in the picture, the strikethrough formatting (which is highly appreciated, great feature!!), is overlapping on the left for task list items which are indeted.

![image](https://user-images.githubusercontent.com/62036826/149196034-de459f62-973b-4bf2-b968-8b207c5b1565.png)

Expected behaviour: The line ends with the checked icon on the left side. 
I know this is just a minor cosmetical bug, but I wanted it to mention it anyway.

### Reproducing

Source code for picture above:

```
- [ ] Anmeldung
    - [x] Mail senden mit Formularen
    - [ ] Antwort vorauss. 17. Januar
```

### Zettlr Version

Stable (most recent version)

### Specify version

2.1.1

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

System: win32 10.0.18363 (architecture: x64)

**Comment by leo-hstone** [comment]
Addition:

![image](https://user-images.githubusercontent.com/62036826/149509583-b6bf5201-b336-4b33-b4db-e6310c1bc4ea.png)

Directly after opening Zettlr, there is no striketrough. Reopening the File doesn't fix it. After I untick and tick one of the boxes, all ticked boxes get updated.

**Comment by nathanlesage** [comment]
> Directly after opening Zettlr, there is no striketrough. Reopening the File doesn't fix it. After I untick and tick one of the boxes, all ticked boxes get updated.

I think I fixed this with 2.1.2, can you confirm this?

**Comment by leo-hstone** [comment]
> can you confirm this?

The strikethroughs were rendered correctly after the update directly on startup.

Unfortunately, the strikethrough is now buggy in comparison to 2.1.1. On my system the strikethrough appears/disappears not until I set the cursor to another positon.

**Comment by kyaso** [comment]
> Unfortunately, the strikethrough is now buggy in comparison to 2.1.1. On my system the strikethrough appears/disappears not until I set the cursor to another positon.

I can confirm this.

**Comment by nathanlesage** [comment]
Huh, that is weird o.O I'll have a look into this. Or you, if you'd like to! Corresponding file is https://github.com/Zettlr/Zettlr/blob/develop/source/common/modules/markdown-editor/hooks/task-item-classes.js

**Comment by kyaso** [comment]
I compared the changes, and found the `cm.on('change',...)` missing.

https://github.com/Zettlr/Zettlr/blob/08c77c3695ad51f30864356e83fff18837ad892a/source/common/modules/markdown-editor/hooks/task-item-classes.js#L20-L38

After adding `cm.on('change', callback)` (below line 37), the checking off works as is used to. Was the missing `change` event the bug (not sure whether the removal of it was intended or not)?

**Comment by nathanlesage** [comment]
Oh wait, then it's missing in the header classes hook as well, since I just copied over that code!

**Comment by derwok** [comment]
BTW. Whoever does not like the whole strike-through UI, and prefers the "old way" of just a ticked-off checkbox, may put this into the Assets / Custom CSS dialog and Save it:

```css
.CodeMirror .task-item-done {
  text-decoration: none;
}
``` 

or this, to make strike-through more subtle (same color code as the blue checkbox and 50% transparent):
```css
.CodeMirror .task-item-done {
  text-decoration-color: rgba(0, 117, 255, 0.5);
}
```

**Comment by nathanlesage** [comment]
@derwok The latter almost seems as if we could put something like that in the codebase …? I also like the strikeout but feel the black line is a tad too big and distorting

**Comment by nathanlesage** [comment]
@kyaso Thanks for your research, I have now added the change event, works better now!

**Comment by nathanlesage** [comment]
Fixed it … somehow. Don't ask me about the motivation of the author of the original mode!

![image](https://user-images.githubusercontent.com/17251683/150006854-75cdbbb0-8a2f-471f-a4cc-071a05a462a8.png)

**Comment by derwok** [comment]
For the interested... With the new CSS from this fix and with Zettlr 2.1.3 onwards, the Custom CSS code to switch off strike-through would be:

```css
.task-item-done .cm-variable-2, .task-item-done .cm-variable-3, .task-item-done .cm-keyword { 
  text-decoration: none !important;
}  
``` 

the code for semi-transparent blue strike-through would be:
```css
.task-item-done .cm-variable-2, .task-item-done .cm-variable-3, .task-item-done .cm-keyword { 
  text-decoration-color: rgba(25,50,200, .5) !important; 
}  
``` 

The "nesting" of CSS rules does not work in the Custom CSS editor. Probably because the code uses Less CSS and the Custom CSS uses vanillia CSS?

**Comment by nathanlesage** [comment]
@derwok exactly, basically you just have to repeat the outer element for every inner element, i.e. take the three classes and prepend `.task-item-done` to each!

**Comment by derwok** [comment]
For the interested... With the new CSS from this fix and with Zettlr 2.1.3 onwards, the Custom CSS code to switch off strike-through would be:

```css
.task-item-done .cm-variable-2, .task-item-done .cm-variable-3, .task-item-done .cm-keyword { 
  text-decoration: none !important;
}  
``` 

the code for semi-transparent blue strike-through would be:
```css
.task-item-done .cm-variable-2, .task-item-done .cm-variable-3, .task-item-done .cm-keyword { 
  text-decoration-color: rgba(25,50,200, .5) !important; 
}  
``` 

The "nesting" of CSS rules does not work in the Custom CSS editor. Probably because the code uses Less CSS and the Custom CSS uses vanillia CSS?

**Comment by Salavin** [comment]
@derwok your custom CSS doesn't appear to work in 2.3.0, presumably because the built-in CSS already has the `!important` tag now for some reason
![image](https://github.com/Zettlr/Zettlr/assets/26843753/bc4cfa44-879d-4221-b3ea-769ce804ed12)
